app/controllers/application_controller.rb

```diff
@@ -30,8 +30,10 @@ class ApplicationController < ActionController::Base
 
   def welcome
     if (model_of_controller.first.nil? rescue false)
-      @welcome = true
-      render :welcome rescue nil
+      if template_exists?(:welcome, _prefixes, variants: request.variant)
+        @welcome = true
+        render :welcome
+      end
     end
   rescue
```

Comment:
I think this `rescue` block wasn't reachable previously (since both conditions that would potentially throw an error had rescue clauses of their own), mind removing it while you're here?

Reply:
Sure :+1: